src/core/public/chrome/nav_group/nav_group_service.ts

```diff
@@ -213,7 +213,7 @@ export class ChromeNavGroupService {
     };
 
     // erase current nav group when switch app don't belongs to any nav group
-    application.currentAppId$.subscribe((appId) => {
+    application.currentAppId$.pipe(debounceTime(500)).subscribe((appId) => {
```

Comment:
Why do you need to debounce?

Reply:
Sure, added some comment.

Reply:
I feel it would be better to have the debounceTime here, so that the consumer that subscribe to `currentNavGroupSorted$` to update UI wouldn't cause the UI flashing when the value changed so frequently.

```typescript
    const currentNavGroupSorted$ = combineLatest([
      this.getSortedNavGroupsMap$(),
      this.currentNavGroup$,
    ])
      .pipe(takeUntil(this.stop$))
      .pipe(debounceTime(500))
      .pipe(
        map(([navGroupsMapSorted, currentNavGroup]) => {
          if (currentNavGroup) {
            return navGroupsMapSorted[currentNavGroup.id];
          }
        })
      );

```

Reply:
Btw, what's the consequence if no `debounceTime(500)`? Anything UI broken?

Reply:
I just removed the reset logic because:
When redirecting to discover, the appId will get changed twice, first is discover the latter is data-explorer. The reset logic will always reset the currentNavGroup to undefined because data-explorer is a hidden application that won't be registered in left navigation in the left navigation.

Reply:
> The effect is that only when user click the `back` button in the new navigation will we reset the currentNavGroup to undefined.

make sense